File: Multiple_Inheritance/2.py
class CalMultiply(object):
    def multiply(self):
        return self.a*self.b

class CalDivide(object):
    def divide(self):
        return self.a/self.b

class Cal(CalMultiply, CalDivide):
    def __init__(self,a,b):
        if isinstance(a, int):
            self.a=a
        if isinstance(b, int):
            self.b=b
    def add(self):
        return self.a+self.b
    def subtract(self):
        return self.a-self.b
    def setA(self, a):
        if isinstance(a,int):
            self.a=a
    def setB(self, b):
        if isinstance(b,int):
            self.b=b
    def getA(self):
        return self.a
    def getB(self):
        return self.b

c = Cal(100, 10)
print(c.add())
print(c.subtract())
print(c.multiply())
print(c.divide())

File: Mixin/2.rb
module CalMultiply
  def multiply
    @a*@b
  end
end
module CalDivide
  def divide
    @a/@b
  end
end
class Cal
  include CalMultiply, CalDivide
  attr_reader :a, :b
  attr_writer :a

  def initialize(a, b)
    @a = a  #인스턴스 변수 = 지역 변수
    @b = b
  end
  def add
    return @a+@b
  end
  def subtract
    return @a-@b
  end
end


c = Cal.new 100,10
p c.add
p c.multiply
p c.divide
